liquidhaskell-boot/src/Language/Haskell/Liquid/LHNameResolution.hs

```diff
@@ -191,11 +192,12 @@ resolveLHNames cfg thisModule localVars impMods globalRdrEnv bareSpec0 dependenc
 
     -- add defines from dependencies to the logical map
     lmap =
-      foldr (\ls lmp ->
-                 lmp <> mkLogicMap (HM.map (fmap lhNameToResolvedSymbol) $ liftedDefines ls)
-            )
-            LH.listLMap $
-            (HM.elems . getDependencies) dependencies
+        (LH.listLMap <>) $
+        mconcat $
+        reverse $
```

Comment:
It doesn't look to me like this `reverse` is necessary? `LogicMap` is internally a hashmap, so the order of addition shouldn't really matter.

Reply:
The preceding `foldr` was careful with the order in which the logic maps were combined. Was it trying to give precedence to some definitions over others perhaps?

Reply:
Nah, the order of the input is non-deterministic anyway. I'll remove the `reverse`.

Reply:
> The preceding `foldr` was careful with the order in which the logic maps were combined. Was it trying to give precedence to some definitions over others perhaps?

Not really, I just put in `foldr` because it's a canonical recursor for lists :)